pkgs/applications/terminal-emulators/rio/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+
+, autoPatchelfHook
+, cmake
+, pkg-config
+, python3
+
+, fontconfig
+, libGL
+, vulkan-loader
+
+, withX11 ? true
+, libX11
+, libXcursor
+, libXi
+, libXrandr
+, libxcb
+
+, withWayland ? false
+, wayland
+}:
+let
+  rlinkLibs = [
+    fontconfig
+    libGL
+    vulkan-loader
+  ] ++ lib.optional withX11 [
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    libxcb
+  ] ++ lib.optional withWayland [
+    wayland
+  ];
+in
+rustPlatform.buildRustPackage rec {
+  pname = "rio";
+  version = "0.0.6.1-unstable-2023-06-13";
+
+  src = fetchFromGitHub {
+    owner = "raphamorim";
+    repo = "rio";
+    rev = "fe44ac3a06cffa56c220688821c8e8eb1cad69b7";
+    hash = "sha256-A8wDv4j48fCdL6Rba2LP6FGIRf7gCa9eK7j7QfUHTsk=";
+  };
+
+  cargoHash = "sha256-laqVqe+4xtz28+Hwh0CVoRyG1Uxxx4Sz67Yy1FOJQcI=";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    cmake
+    pkg-config
+    python3
+  ];
+
+  autoPatchelfIgnoreMissingDeps = [
+    "libgcc_s.so.1"
```

Comment:
Why you're ignoring this?

Reply:
The hook fails otherwise.

Reply:
You need to add whatever dependency has this library instead of ignoring, unless you're ignoring for a good reason.

Reply:
```
$ nix-locate --top-level libgcc_s.so.1
remarkable2-toolchain.out                       120,392 r /nix/store/5aalvma9rm7h6k9892dd37h2n20dn943-remarkable2-toolchain-3.1.2/sysroots/cortexa7hf-neon-remarkable-linux-gnueabi/lib/libgcc_s.so.1
remarkable2-toolchain.out                       100,696 r /nix/store/5aalvma9rm7h6k9892dd37h2n20dn943-remarkable2-toolchain-3.1.2/sysroots/x86_64-codexsdk-linux/lib/libgcc_s.so.1
remarkable-toolchain.out                        116,300 r /nix/store/0wxz8w44xmy689fh2fi62r3xrrw0mqp6-remarkable-toolchain-3.1.2/sysroots/cortexa9hf-neon-remarkable-linux-gnueabi/lib/libgcc_s.so.1
remarkable-toolchain.out                        100,696 r /nix/store/0wxz8w44xmy689fh2fi62r3xrrw0mqp6-remarkable-toolchain-3.1.2/sysroots/x86_64-codexsdk-linux/lib/libgcc_s.so.1
julia_18-bin.out                                578,744 x /nix/store/shk9wn0a1gdcb59nyr4clfy6srkpz8xr-julia-bin-1.8.5/lib/julia/libgcc_s.so.1
julia.out                                       141,696 x /nix/store/4jy2gcc6hcpzj61z984yn7pa27zy4scl-julia-1.9.1/lib/julia/libgcc_s.so.1
julia_18.out                                    141,696 x /nix/store/31awcfcy615yr32hxyzq24dkzyryk2b5-julia-1.8.5/lib/julia/libgcc_s.so.1
julia-lts.out                                   758,680 x /nix/store/njwlrj5r130czsxvhxwfmn0yhg1njxlj-julia-bin-1.6.6/lib/julia/libgcc_s.so.1
julia-bin.out                                   578,744 x /nix/store/wfbk3pn2clcaagmcbp74ana4djngnp06-julia-bin-1.9.1/lib/julia/libgcc_s.so.1
gcc-unwrapped.libgcc                            141,696 r /nix/store/v75y3qd2i450dgpmd2i9qk9vwmn0qmrh-gcc-12.3.0-libgcc/lib/libgcc_s.so.1
gcc-unwrapped.lib                                     0 s /nix/store/n7pvb7gdf1g6dvj7sl92i882qjl4kyx9-gcc-12.3.0-lib/lib/libgcc_s.so.1
```

So it looks like you need to add `gcc-unwrapped.libgcc` to the `buildInputs`.

Reply:
Fixed. Thanks for the hint.